lang/C.md

```diff
@@ -0,0 +1,112 @@
+    return 0;
+}
+```
+## What is C?
+
+C programming is a general-purpose, procedural, imperative computer programming language developed in 1972 by Dennis M. Ritchie at the Bell Telephone Laboratories to develop the UNIX operating system. C programming is considered as the base for other programming languages, that is why it is known as mother language. C language is a system programming language because it can be used to do low-level programming (for example driver and kernel). It is generally used to create hardware devices, OS, drivers, kernels, etc. For example, Linux kernel is written in C. It supports the feature of both low-level and high-level languages.
+
+## Why one should use C?
+
+- Easy to learn
+- Structured language
+- It produces efficient programs
+- It can handle low-level activities
+- It can be compiled on a variety of computer platforms
+
+C programming language has fewer libraries in comparison with other high-level languages. So, learning C programming also clears programming concepts to a great extent as you have to write lot of things from scratch. You will not be dependent on the programming language entirely for implementing some basic operations and implementing them on your own will also help to build up your analytical skills. C is extensively used in Embedded Programming. Embedded Programming is also referred to as micro-controller programming, where C program is used to control micro-controllers. Microcontrollers and embedded programming is widely used in auto-motives, Robotics, Hardwares etc.
+
+## Setting up C environment locally
+
+**Step 1:**
+
+You have to download a text editor or an IDE to type your code. Examples include Notepad++,Sublime,vim. An integrated development environment or an IDE is a software application that provides comprehensive facilities to computer programmers for software development. Examples include Visual Studio Code, CLion.
+
+- Sublime - Navigate to [https://www.sublimetext.com/3](https://www.sublimetext.com/3) to download the latest stable version of Sublime according to your OS (Windows/Linux/OSX).
+- Visual Studio Code - Navigate to [https://code.visualstudio.com/download](https://code.visualstudio.com/download)  to download the latest stable version of Sublime according to your OS (Windows/Linux/OSX).
+
+**Step 2:**
+
+Install the Compiler according to your OS
+
+#### Linux   
+
+If you are using Linux , then check whether GCC is installed on your system by entering the following command from the command line - `gcc --version`
+
+If you have GNU compiler installed on your machine, then it should print a message as follows:
+
+"gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0 Copyright (C) 2019 Free Software Foundation, Inc. 
+This is free software; see the source for copying conditions.  There is NO
+warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE."
+
+If it is not installed you can install using these commands:
+
+`sudo apt-get install gcc`
```

Comment:
Wrap code blocks in 3 backticks followed by a line brake, to render them as syntax highlighted code blocks on the frontend.

Here's an example replacement 👇 
<pre><code>```bash
sudo apt-get install gcc
```</code></pre>

Maybe 🤷‍♂️ even replace `apt-get` with `apt` as `apt-get` is no longer the recommended way of interacting with the package manager, although so many tutorials on the internet are still using it.